Feature/add typo3 sniffs from old package

justusmoroni opened the pull request.

Commit:
TASK: Add typo3sniffpool

We need the TYPO3 sniffs.

Commit:
TASK: Add old typo3 rules

Add old rules from typo3-ci/typo3cms package which
was deleted.

Commit:
TASK: Change name of cgl standard

Add documentation for installation.
Change name of package.

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <!-- Exclude third party code -->
+    <exclude-pattern>*/.git/*</exclude-pattern>
+    <exclude-pattern>*typo3/contrib/*</exclude-pattern>
+    <exclude-pattern>*typo3/sysext/adodb/adodb/*</exclude-pattern>
```
Why do we not exclude the whole `typo3` folder but some sub-sub folders and files?

Or even, instead of excluding so mach, just lint the stuff we want, which should basically just be `localPackages` and some configuration files like `AdditionalConfiguration.php`?

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/FileCommentSniff.php" />
+    <rule ref="TYPO3SniffPool.Commenting.FileComment">
+        <exclude-pattern>ext_emconf.php</exclude-pattern>
+        <exclude-pattern>ext_tables.php</exclude-pattern>
```
Shouldn't we add `ext_localconf.php`?

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+
+    <!-- Debug -->
+    <rule ref="../TYPO3SniffPool/Sniffs/Debug/DebugCodeSniff.php">
+        <exclude-pattern>*typo3/sysext/core/Classes/Utility/DebugUtility.php</exclude-pattern>
```
We should not need any exclude here, as we should only lint our own files, see comment above.

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+
+    <!-- Files -->
+    <rule ref="Generic.Files.OneClassPerFile">
+        <exclude-pattern>*typo3/sysext/core/Migrations/Code/LegacyClassesForIde.php</exclude-pattern>
```
We should not need any exclude here, as we should only lint our own files, see comment above.

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <rule ref="Generic.Files.LineEndings" />
+    <rule ref="Generic.Files.LineLength">
+        <properties>
+            <property name="lineLimit" value="130"/>
```
Looks a bit long to me, should we use that high values?

justusmoroni replied:
I've removed it and use the default of the PSR-2 standard:
```
<!-- The soft limit on line length MUST be 120 characters; automated style checkers MUST warn but MUST NOT error at the soft limit. -->
 <rule ref="Generic.Files.LineLength">
  <properties>
   <property name="lineLimit" value="120"/>
   <property name="absoluteLineLimit" value="0"/>
  </properties>
 </rule>
```

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <rule ref="Generic.PHP.UpperCaseConstant" />
+    <rule ref="Generic.PHP.DeprecatedFunctions" />
+    <rule ref="Squiz.PHP.Eval">
+        <exclude-pattern>*/tests/*</exclude-pattern>
```
We should not need any exclude here, as we should only lint our own files, see comment above.

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <!-- Scope -->
+    <rule ref="Squiz.Scope.MemberVarScope" />
+    <rule ref="Squiz.Scope.MethodScope" />
+    <!--<rule ref="../TYPO3SniffPool/Sniffs/Scope/AlwaysReturnSniff.php" />-->
```
Should we remove this commented configuration?

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <rule ref="Squiz.WhiteSpace.OperatorSpacing" />
+    <rule ref="../TYPO3SniffPool/Sniffs/WhiteSpace/NoWhitespaceAtInDecrementSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/WhiteSpace/DisallowSpaceIndentSniff.php" />
+    <!--<rule ref="Generic.WhiteSpace.DisallowSpaceIndentSniff.php" />-->
```
Should we remove this commented configuration?

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,137 @@
+    <rule ref="Squiz.WhiteSpace.SuperfluousWhitespace" />
+    <rule ref="Squiz.WhiteSpace.OperatorSpacing" />
+    <rule ref="../TYPO3SniffPool/Sniffs/WhiteSpace/NoWhitespaceAtInDecrementSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/WhiteSpace/DisallowSpaceIndentSniff.php" />
```
We always use spaces for indentation, looks like this sniff should be excluded instead.

Should we not include the complete `TYPO3SniffPool` and afterwards exclude which we do not need?

Commit:
TASK: Add our custom typo3 rules

Some of the rules were old.
TYPO3 uses PSR-2 as basis now.
Add our own rules and excludes.

DanielSiepmann commented on readme.rst:
```diff
@@ -1,6 +1,20 @@
+------------
+
+You have to require the package via composer `composer require codappix/cgl-typo3`.
+After that you have to add the following to the `composer.json`:
```
Shouldn't this be added before? Otherwise we should point out that you have to run `composer dumpautoload` afterwards.

DanielSiepmann replied:
Shouldn't we use a code-block? Currently the `composer.json` part is rendered as quote, right?

DanielSiepmann commented on ruleset.xml:
```diff
@@ -1,5 +1,164 @@
+    <rule ref="Generic.Files.LineLength" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Files/IncludingFileSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Files/FilenameSniff.php">
+        <!-- Don't check for PHPDoc in this files -->
```
PHPDoc? This rule checks the file name. Looks like the comment is wrong.
